### PR TITLE
Add IncrementalPostProcessorHost, to support load and save context info

### PR DESCRIPTION
**Pull request by hellosnow**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/PostProcessors/IncrementalPostProcessorHost.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+            return File.OpenRead(Path.Combine(_increContext.LastBaseDir, lastPostProcessorInfo.ContextInfoFile));
+        }
+
+        public Stream SaveContextInfo()
+        {
+            var currentPostProcessorInfo = FindPostProcessorInfo(_increContext.CurrentInfo, _postProcessorName);
+            currentPostProcessorInfo.ContextInfoFile = IncrementalUtility.CreateRandomFileName(_increContext.CurrentBaseDir);
+
+            return File.OpenWrite(Path.Combine(_increContext.CurrentBaseDir, currentPostProcessorInfo.ContextInfoFile));
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/PostProcessors/IncrementalPostProcessorHost.cs]: File.Create

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Plugins/IPostProcessorHost.cs [hunk 2]
```diff
@@ -3,7 +3,20 @@
...
 namespace Microsoft.DocAsCode.Plugins
 {
+    using System.IO;
+
     public interface IPostProcessorHost
     {
+        /// <summary>
+        /// Load context info from last post processing.
+        /// </summary>
+        /// <returns>Stream to be read, Return null when there is no last info.</returns>
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Plugins/IPostProcessorHost.cs]: >R [](start = 40, length = 1)

minor: r

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by vwxyzh** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**